process/block/poolsCleaner/basePoolsCleaner.go

```diff
@@ -22,13 +24,16 @@ type basePoolsCleaner struct {
 	shardCoordinator               sharding.Coordinator
 	maxRoundsToKeepUnprocessedData int64
 	cancelFunc                     func()
+	isCleaningRoutineRunning       atomic.Flag
```

Comment:
since you have a mutex protection, could have been a plain old bool var. Can be left as it is

Reply:
pushed